config/config.go

```diff
@@ -282,6 +284,23 @@ func (rc *ReporterConfig) NewReporter(
 	return reporter, err
 }
 
+// To permit clean testing.
+var newUDPTransport = jaeger.NewUDPTransport
```

Comment:
can we do without this? We already have tests and they didn't need this.

Reply:
Maybe: I don't know enough go. Here's the problem.

There are two existing places we could put this code. We could put it in NewUDPTransport, but that seems to make it less general than it should be - and the Config should do the logic, so that we don't have all the different environment loookups scattered around (which is what will happen if we put it somewhere else).

But, Config creates NewUDPTransport directly, and the transport doesn't store the host:port it was created with, so we can't introspect it to verify that the right transport was created.

I guess we could just depend on functional tests, but that seems like a lot of the stack to use when we just want to know that the variable was looked up and passed to the constructor.

Reply:
Maybe we should adopt the same model as in Jaeger SDK for Java - the Config class has `initFromEnv()` function that reads env and populates the config itself. This way you don't mix up env-reading code with the actual object creation.

Reply:
Ok, I'll refresh and align with that